src/Microsoft.IdentityModel.Tokens/Validation/Validators.Audience.cs

```diff
@@ -75,106 +69,59 @@ internal static AudienceValidationResult ValidateAudience(IEnumerable<string> au
                         typeof(SecurityTokenInvalidAudienceException),
                         new StackFrame(true)));
 
-            if (string.IsNullOrWhiteSpace(validationParameters.ValidAudience) && (validationParameters.ValidAudiences == null))
+            if (audiences.Count == 0)
                 return new AudienceValidationResult(
                     Utility.SerializeAsSingleCommaDelimitedString(audiences),
                     ValidationFailureType.NullArgument,
-                    new ExceptionDetail(
-                        new MessageDetail(
-                            LogMessages.IDX10208,
-                            null),
-                        typeof(SecurityTokenInvalidAudienceException),
-                        new StackFrame(true)));
-
-            if (audiences is not List<string> audiencesAsList)
-                audiencesAsList = audiences.ToList();
-
-            if (audiencesAsList.Count == 0)
-                return new AudienceValidationResult(
-                    Utility.SerializeAsSingleCommaDelimitedString(audiencesAsList),
-                    ValidationFailureType.NullArgument,
                     new ExceptionDetail(
                         new MessageDetail(
                             LogMessages.IDX10206,
                             null),
                         typeof(SecurityTokenInvalidAudienceException),
                         new StackFrame(true)));
 
-            string? validAudience = AudienceIsValidReturning(audiencesAsList, validationParameters);
+            string? validAudience = AudienceIsValidReturning((List<string>)audiences, validationParameters);
```

Comment:
We do not need to call AudienceIsValidReturning.
Call directly into AudienceMatchList.